### PR TITLE
Proper assignment of quest loot for group loot & round robin

### PR DESCRIPTION
**Pull request by rennr**
**Changes proposed:**

Quest items for group loot and round robin loot do not work properly in the current implementation. Party members should be able to take turns looting quest items if using group loot or round robin. Current functionality is that nearly all quest items are free for all. This is not Blizzlike.

The proposed patch makes the following changes:

-  In group loot or round robin, only the person who owns the loot sees and can loot the quest items
-  If the looter releases the loot, the quest items become available to everyone
-  If the quest items are multi-loot (i.e., everyone gets the quest item), then it will appear for everyone as expected
-  If the loot type is any other type (e.g., master looter, free for all), the previous functionality takes effect (i.e., quest loot is free for all in most instances)

**Tests performed:**

Builds. Tested in-game.

### CODE REVIEW

**Comment by CraftedRO** [comment]
https://github.com/TrinityCore/TrinityCore/issues/22829 this might be related

**Comment by rennr** [comment]
> #22829 this might be related

Yeah that is the exact issue this pull request resolves. I've been playing with my version of the code since a few days before I submitted this PR and so far there have been no issues - it works as I outlined above, which is how it was in WotLK retail.

**Comment by CraftedRO** [comment]
71b7cc6361d6 Just tested this and seems fine it does fix the above mentioned issue:


https://user-images.githubusercontent.com/24683355/233100618-4570ab2d-dfad-48f7-b64a-72541c8c3d9d.mp4

**Comment by rennr** [comment]
What else is needed for this to be merged?

File: src/server/game/Loot/Loot.cpp [hunk 1]
```diff
@@ -86,6 +96,9 @@ bool LootItem::AllowedForPlayer(Player const* player, bool isGivenByMasterLooter
     if (pProto->Class == ITEM_CLASS_RECIPE && pProto->Bonding == BIND_WHEN_PICKED_UP && pProto->Spells[1].SpellId != 0 && player->HasSpell(pProto->Spells[1].SpellId))
         return false;
 
+    if (needs_quest && !freeforall && player->GetGroup() && (player->GetGroup()->GetLootMethod() == GROUP_LOOT || player->GetGroup()->GetLootMethod() == ROUND_ROBIN) && ownerGuid.IsEmpty() == false && ownerGuid != player->GetGUID())
```
Comment (jackpoz) [inline src/server/game/Loot/Loot.cpp:99]:
```suggestion
    if (needs_quest && !freeforall && player->GetGroup() && (player->GetGroup()->GetLootMethod() == GROUP_LOOT || player->GetGroup()->GetLootMethod() == ROUND_ROBIN) && !ownerGuid.IsEmpty() && ownerGuid != player->GetGUID())
```

**Comment by jackpoz** [comment]
>Tests performed:
Builds. Tested in-game.

This PR changes quite a few places related to the loot, what kind of tests did you do ?¨

I added a small codestyle change to replace a == false check with ! .

**Comment by rennr** [comment]
> > Tests performed:
> > Builds. Tested in-game.
> 
> This PR changes quite a few places related to the loot, what kind of tests did you do ?
> 
> I added a small codestyle change to replace a == false check with ! .

I've played several hours with others with these changes and there were no issues that I experienced. The code changes cause the expected results and there have been no side effects that I've encountered.

**Comment by jackpoz** [comment]
@rennr this PR has been merged, please keep an eye on new issues if they might be related to this.

Thanks for the PR 🚀 😃

**Pull request merged**